conductor-client/src/main/java/com/openlattice/authorization/HazelcastAclKeyReservationService.java

```diff
@@ -193,6 +206,60 @@ public void reserveIdAndValidateType( EntitySet entitySet ) {
         }
     }
 
+    /**
+     * Simple reservation algorithm named securable objects. It finds a free id and then either returns the actual
+     * id of the object (as reserved by another caller).
+     *
+     * This routine is simpler, because it doesn't have to worry about other calls generating conflicting ids for the
+     * same name. It should be used whenever the name of the object is known before the object is created.
+     *
+     * @param name The name to reservation.
+     * @return A UUID for the name reservation.
+     */
+    public UUID reserveOrGetId( String name ) {
+        final UUID maybeActualId;
+        final UUID generatedId;
+
+        try {
+            aclKeys.lock( name, 1, TimeUnit.MINUTES );
+            generatedId = HazelcastUtils.insertIntoUnusedKey( names, name, UUID::randomUUID );
+            maybeActualId = aclKeys.putIfAbsent( name, generatedId );
+        } finally {
+            aclKeys.unlock( name );
+        }
+
+        if ( maybeActualId == null ) {
+            return generatedId;
+        } else {
+            names.delete( generatedId );
+            return maybeActualId;
+        }
+    }
+
+    public <T extends AbstractSecurableObject> T reserveOrGetId(
+            T aso,
+            Function<AbstractSecurableObject, String> nameGen ) {
+        final var name = nameGen.apply( aso );
+        var id = aso.getId();
+
+        //Make sure ID is free.
+        while ( names.putIfAbsent( id, name ) != null ) {
+            id = UUID.randomUUID();
+        }
+
+        //Make sure name hasn't been assigned another id. If it has then return that id and free currently assigned id.
+        final var maybeActualId = aclKeys.putIfAbsent( nameGen.apply( aso ), aso.getId() );
```

Comment:
You already did `nameGen.apply( aso )` and `aso.getId() ` above. Can you use `name` and `id` here instead of doing `.apply()` and `.getId()` again?

```suggestion
        final var maybeActualId = aclKeys.putIfAbsent( name, id );
```